app/src/main/java/com/github/se/polyfit/data/api/SpoonacularAPI.kt

```diff
@@ -0,0 +1,106 @@
+    return imageAnalysis(file)
+  }
+
+  /**
+   * Analyze a food image. The API tries to classify the image, guess the nutrition, and find a
+   * matching recipes.
+   *
+   * @param file The image file to classify
+   * @return The response from the API
+   */
+  fun imageAnalysis(file: File): ImageAnalysisResponseAPI {
+    val mediaType = "multipart/form-data".toMediaTypeOrNull()
+    val requestBody = file.asRequestBody(mediaType)
+    val body =
+        MultipartBody.Builder()
+            .setType(MultipartBody.FORM)
+            .addFormDataPart("file", file.name, requestBody)
+            .build()
+
+    val request =
+        Request.Builder()
+            .url(API_URL + IMAGE_ANALYSIS_ENDPOINT)
+            .post(body)
+            .addHeader("content-type", "multipart/form-data; boundary=---011000010111000001101001")
+            .addHeader("X-RapidAPI-Key", BuildConfig.X_RapidAPI_Key)
+            .addHeader("X-RapidAPI-Host", BuildConfig.X_RapidAPI_Host)
+            .build()
+
+    return try {
+      Log.e(
+          "SpoonacularApiCaller",
+          "Sending image analysis request to ${API_URL + IMAGE_ANALYSIS_ENDPOINT}")
+      val response = client.newCall(request).execute()
+
+      if (!response.isSuccessful) {
+        Log.e("SpoonacularApiCaller", "Error during image analysis: ${response.code}")
+        throw Exception("Error during image analysis: ${response.code}")
+      }
+      val responseBody = response.body?.string() ?: ""
+      val jsonObject = JSONObject(responseBody)
+      ImageAnalysisResponseAPI.fromJsonObject(jsonObject)
+    } catch (e: IOException) {
+      Log.e("SpoonacularApiCaller", "Error during image analysis", e)
+      throw Exception("Error during image analysis : ", e)
+    }
+  }
+
+  /**
+   * Get the nutritional information for a recipe.
+   *
+   * @param recipeId The ID of the recipe
+   * @return The response from the API
+   */
+  fun getRecipeNutrition(recipeId: Int): RecipeNutritionResponseAPI {
+    val request =
+        Request.Builder()
+            .url(API_URL + RECIPE_NUTRITION_ENDPOINT.format(recipeId))
+            .get()
+            .addHeader("X-RapidAPI-Key", BuildConfig.X_RapidAPI_Key)
+            .addHeader("X-RapidAPI-Host", BuildConfig.X_RapidAPI_Host)
+            .build()
+
+    return try {
+      val response = client.newCall(request).execute()
+
+      val requestStatus = response.code
+
+      if (requestStatus != 200) {
+        Log.e("SpoonacularApiCaller", "Error getting recipe nutrition: $requestStatus")
+        throw Exception("Error getting recipe nutrition: $requestStatus")
+      }
+
+      val responseBody = response.body?.string() ?: ""
+      val jsonObject = JSONObject(responseBody)
+      RecipeNutritionResponseAPI.fromJsonObject(jsonObject)
+    } catch (e: IOException) {
+      Log.e("SpoonacularApiCaller", "Error getting recipe nutrition", e)
+      throw Exception("Error getting recipe nutrition : ", e)
+    }
+  }
+}
```

Comment:
seems good, I don't find the API key so ensure there's no API key in vc

Reply:
Yeah that is normal. The api key is stored in the local.properties which is not push to the git. I will send the keys  in the group chat when this implementation is accepted to that everyone can copy them locally.